### ISSUE TITLE
Strange behavior happens when I execute a spark shell on a remote machine

### ISSUE THREAD

**Issue by EugeneChung** [warpdotdev/Warp#5527]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

When I get into a remote machine with ssh and execute a spark shell on the machine, strange characters are automatically inserted without stop and the spark shell is shut down.
<img width="1046" alt="image" src="https://github.com/user-attachments/assets/cbe487db-2539-4ab9-83e9-3544d2874b9e">
<img width="1460" alt="image" src="https://github.com/user-attachments/assets/154d1f72-68ff-43ae-88cf-ae972c632856">
<img width="1510" alt="image" src="https://github.com/user-attachments/assets/9d9246a9-990b-4940-8056-e5af6cdeb847">

### To reproduce

1. ssh to my remote machine
2. `spark-shell --master yarn --driver-memory 2G --executor-memory 2G --executor-cores 2 --num-executors 2 \
--conf spark.dynamicAllocation.enabled=false`

### Expected behavior

- scala shell on my PC using warp
<img width="492" alt="image" src="https://github.com/user-attachments/assets/d45b2518-7a38-465f-9c6d-0e9c93a9cc8c">

- spark shell (scala) on a remote maching using iterm
<img width="1239" alt="image" src="https://github.com/user-attachments/assets/c545ba09-f41d-4f0b-afde-f9d962fc5eea">

### Operating system

MacOS

### Operating system and version

Sonoma 14.7.1

### Shell Version

GNU bash, version 5.2.15(1)-release (aarch64-amazon-linux-gnu)

### Current Warp version

v0.2024.11.19.08.02.stable_03

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Recent working Warp date

N/A

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this an issue only in Warp?

Yes, I confirmed that this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by EugeneChung** [comment]
It's okay with pyspark on the same remote machine. Only the Scala shell is a problem.
<img width="844" alt="image" src="https://github.com/user-attachments/assets/f56ca788-09e7-4a1a-809d-4b5c5b884db0">

**Comment by EugeneChung** [comment]
I don't know why but I can't reproduce right now....

**Comment by EugeneChung** [comment]
I've found the reproduction scenario.
`Split pane right CMD-D` then do ssh & spark-shell.

- Here is the full debug log. From the line 1715 `10:20:43 [DEBUG] Received DCS string`, I did 'Split pane right'. Then from the line 1861, `10:21:05 [WARN] redraw_frame was called 2 times before the frame was drawn` strange behaviour started.
[warp-github-issue-5527-debug-logs.txt](https://github.com/user-attachments/files/17985862/warp-github-issue-5527-debug-logs.txt)
